java/src/main/java/ai/rapids/cudf/Aggregation.java

```diff
@@ -300,7 +300,11 @@ public boolean equals(Object other) {
         }
     }
 
-    private static final class CollectSetAggregation extends Aggregation {
+    /**
+     * WARNING: https://github.com/rapidsai/cudf/issues/8405
+     */
+    public static final class CollectSetAggregation extends Aggregation
```

Comment:
Isn't the class `private ` here ? Since this class should not be used directly, instead calling `collectSet` to get one.

Reply:
It should be `public` because we need to use this class as a type parameter in spark-rapids, just like [here](https://github.com/NVIDIA/spark-rapids/blob/branch-21.06/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala#L670).

Reply:
OK, if the class is required to be public, shall we need to make its constrcutors be private to avoid creating a CollectSetAggregation directly ?

Reply:
Good idea, it's private now.